File: app/src/main/java/com/wordpress/lonelytripblog/funwithflags/data/GameRepo.java
package com.wordpress.lonelytripblog.funwithflags.data;

import androidx.lifecycle.LiveData;

import com.wordpress.lonelytripblog.funwithflags.data.db.Country;

/**
 * Created by Павел on 09.03.2018.
 */

public interface GameRepo {
    LiveData<GameEntity> getLiveDataForGame();
    void nextFlag();
    void saveCurrentFlagIntoLearntFlags();
    LiveData<Integer> getAmountOfLeftFlags();
    LiveData<Country> getLearntFlag();
    void nextLearntFlag();
    LiveData<Integer> getAmountOfLearntFlags();
}

File: app/src/main/java/com/wordpress/lonelytripblog/funwithflags/data/GameRepo.kt
package com.wordpress.lonelytripblog.funwithflags.data

import androidx.lifecycle.LiveData
import com.wordpress.lonelytripblog.funwithflags.data.db.Country

interface GameRepo {
    fun getUnknownCountryGameEntity(): LiveData<GameEntity>
    fun requestNewGameEntity()
    fun saveCurrentFlagIntoLearntFlags()
    fun getAmountOfLeftFlags(): LiveData<Int>
    fun getLearntFlags(): LiveData<List<Country>>
    fun getAmountOfLearntFlags(): LiveData<Int>
}